diff --git a/linker/linker.cpp b/linker/linker.cpp
index 7909241..19f461c 100644
--- a/linker/linker.cpp
+++ b/linker/linker.cpp
@@ -3260,7 +3260,6 @@ static ElfW(Addr) __linker_init_post_relocation(KernelArgumentBlock& args, ElfW(
   ElfW(Ehdr)* elf_hdr = reinterpret_cast<ElfW(Ehdr)*>(si->base);
   if (elf_hdr->e_type != ET_DYN) {
     __libc_format_fd(2, "error: only position independent executables (PIE) are supported.\n");
-    exit(EXIT_FAILURE);
   }
 #endif
 
